pkg/frontend/admin_vm_actions.go

```diff
@@ -0,0 +1,60 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+)
+
+type azVmActionsWrapper struct {
+	vmName       string
+	adminActions adminactions.AzureActions
+	doc          *api.OpenShiftClusterDocument
+	vars         map[string]string
+}
+
+func (f *frontend) buildAzureActionsFactory(ctx context.Context, r *http.Request, log *logrus.Entry) (azVmActionsWrapper, error) {
+	vars := mux.Vars(r)
+
+	vmName := r.URL.Query().Get("vmName")
+	err := validateAdminVMName(vmName)
+	if err != nil {
+		return azVmActionsWrapper{}, err
+	}
+
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return azVmActionsWrapper{}, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
```

Comment:
for another PR (but it's also DRY :D), could this newclouderror be made into something like `api.NewCloudErrorResourceNotFound(map[string]string)` or something ? it appears  16 times over the codebase with this exact wording(well maybe 12 after that PR). Maybe I missed some also

Reply:
I started working on a fix for this separately since it is a significant refactor itself.

Reply:
Awesome thanks !